info/available-rules.md

```diff
@@ -1,131 +1,131 @@
+| 3    | 3.1.2    | FILE_NO_BLANK_LINE_BETWEEN_BLOCKS         | Check: warns if there is not exactly one blank line between the code parts.<br>Fix: leaves a single empty line                                                                                                                                                     | yes | no                                                                                                                                                                                                                         | -                                                                                                                                                                                                                             |
+| 3    | 3.1.2    | FILE_UNORDERED_IMPORTS                    | Check: warns if the imports are not sorted alphabetically or contain empty lines among them<br>Fix: reorders imports.                                                                                                                                            | yes | no                                                                                                                                                                                                                         | -                                                                                                                                                                                                                             |
+| 3    | 3.1.2    | FILE_WILDCARD_IMPORTS                     | Check: warns if the wildcard imports are used except allows.                                                                                                                                                                                                     | no  | allowedWildcards                                                                                                                                                                                                           | -                                                                                                                                                                                                                             |
```

Comment:
"Check: warns if the wildcard imports are used except allows." Please explain.

Reply:
This means that the so-called [wildcard imports](https://kotlinlang.org/spec/packages-and-imports.html), while accepted by the compiler, should be prohibited as per the _Kotlin Style Guide_, and any violations except for, well, exceptions, should be reported by _diKTat_ (as a part of `FILE_WILDCARD_IMPORTS` rule implementation).

If you look at `diktat-analysis.yml`, it adds a possiblility to configure `allowedWildcards`:

```yml
# Checks that there is no wildcard imports. Exception: allowedWildcards
- name: FILE_WILDCARD_IMPORTS
  enabled: true
  configuration:
    # Allowed wildcards for imports (e.g. "import org.cqfn.diktat.*, import org.jetbrains.kotlin.*")
    allowedWildcards: "kotlinx.serialization.*,org.cqfn.diktat.ruleset.utils.*"
```

Reply:
OK. I changed it as follows then:

Check: warns if the wildcard imports are used except allow**ed.**

Reply:
Or:
Check: warns if the wildcard imports are used except the cases when they are allowed.